client/src/main/java/org/apache/rocketmq/client/trace/AsyncTraceDispatcher.java

```diff
@@ -328,13 +330,12 @@ public void addTraceTransferBean(TraceTransferBean traceTransferBean) {
             initFirstBeanAddTime();
             this.traceTransferBeanList.add(traceTransferBean);
             this.currentMsgSize += traceTransferBean.getTransData().length();
-            if (currentMsgSize >= traceProducer.getMaxMessageSize() - 10 * 1000) {
+            traceTransferBean.getTransKey().forEach(x -> this.currentMsgKeySize += x.length());
```

Comment:
[reduce](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#reduce-T-java.util.function.BinaryOperator-), IMO, is a better fit here.

Reply:
> Will `reduce` be a better fit than `forEach`?

Is it modified like this `this.currentMsgKeySize = traceTransferBean.getTransKey().stream()
                .reduce(currentMsgKeySize, (acc, x) -> acc + x.length(), Integer::sum);`? I feel even more complicated.

Reply:
But JVM may potentially vectorize this loop, making SIMD possible